cognite_toolkit/_cdf_tk/storageio/selectors/_instances.py

```diff
@@ -3,13 +3,38 @@
 from pathlib import Path
 from typing import Literal
 
-from cognite.client.data_classes.data_modeling import EdgeId, NodeId
+from cognite.client.data_classes.data_modeling import EdgeId, NodeId, ViewId
 from cognite.client.utils._identifier import InstanceId
+from pydantic import Field
 
-from cognite_toolkit._cdf_tk.resource_classes.views import ViewReference
+from cognite_toolkit._cdf_tk.constants import DM_EXTERNAL_ID_PATTERN, DM_VERSION_PATTERN, SPACE_FORMAT_PATTERN
 from cognite_toolkit._cdf_tk.storageio._data_classes import InstanceIdCSVList
 
-from ._base import DataSelector
+from ._base import DataSelector, SelectorObject
+
+
+class SelectedView(SelectorObject):
```

Comment:
We stop using ViewReference as that is not immutable and thus not hashable. In addition, we want to make version optional.